gluten-core/src/main/scala/org/apache/spark/sql/internal/SparkConfigUtil.scala

```diff
@@ -50,15 +50,15 @@ object SparkConfigUtil {
   }
 
   def get[T](conf: SparkConf, entry: ConfigEntry[T]): T = {
-    entry.valueConverter(conf.get(entry.key, entry.defaultValueString))
+    entry.valueConverter(checkUndefined(conf.get(entry.key, entry.defaultValueString)))
```

Comment:
Instead of calling `checkUndefined` every time, we can modify like this
```diff
-    entry.valueConverter(conf.get(entry.key, entry.defaultValueString))
+    conf
+      .getOption(entry.key)
+      .map(entry.valueConverter)
+      .getOrElse(entry.defaultValue.getOrElse(None).asInstanceOf[T])
```

Reply:
@yaooqinn Should we also apply `entry.valueConverter` to the defaultValue? So it will be
```scala
entry.valueConverter(
      conf
        .getOption(entry.key)
        .getOrElse(entry.stringConverter(entry.defaultValue.getOrElse(None).asInstanceOf[T])))
```
or
```scala
entry.valueConverter(
      conf
        .getOption(entry.key)
        .getOrElse(checkUndefined(entry.defaultValueString)))
```
And for `conf: java.util.Map[String, String]`， it will be
```scala
entry.valueConverter(
      Option(conf.get(entry.key))
        .getOrElse(checkUndefined(entry.defaultValueString)))
```
What do you think?

Reply:
What `valueConverter + checkUndefined + defaultValueString` produces is `defaultValue`. So, how about we use it directly?